Src/PerformanceCollector/Filtering/DocumentStream.cs

```diff
@@ -0,0 +1,239 @@
+                    ////    CollectionConfigurationError.CreateError(
+                    ////        CollectionConfigurationErrorType.DocumentStreamFilterFailureToRun,
+                    ////        string.Format(CultureInfo.InvariantCulture, "Document stream filter failed to run"),
+                    ////        e));
+                    conjunctionGroupPassed = false;
+                }
+
+                if (conjunctionGroupPassed)
+                {
+                    // no need to check remaining groups, one OR-connected group has passed
+                    leastOneConjunctionGroupPassed = true;
+                    break;
+                }
+            }
+
+            errors = errorList.ToArray();
+
+            return leastOneConjunctionGroupPassed;
+        }
+
+        private void CreateFilters(out CollectionConfigurationError[] errors)
+        {
+            var errorList = new List<CollectionConfigurationError>();
+            foreach (DocumentFilterConjunctionGroupInfo documentFilterConjunctionGroupInfo in this.info.DocumentFilterGroups ?? new DocumentFilterConjunctionGroupInfo[0])
+            {
+                try
+                {
+                    CollectionConfigurationError[] groupErrors;
+                    switch (documentFilterConjunctionGroupInfo.TelemetryType)
+                    {
+                        case TelemetryType.Request:
+                            this.requestFilterGroups.Add(new FilterConjunctionGroup<RequestTelemetry>(documentFilterConjunctionGroupInfo.Filters, out groupErrors));
+                            break;
+                        case TelemetryType.Dependency:
+                            this.dependencyFilterGroups.Add(new FilterConjunctionGroup<DependencyTelemetry>(documentFilterConjunctionGroupInfo.Filters, out groupErrors));
+                            break;
+                        case TelemetryType.Exception:
+                            this.exceptionFilterGroups.Add(new FilterConjunctionGroup<ExceptionTelemetry>(documentFilterConjunctionGroupInfo.Filters, out groupErrors));
+                            break;
+                        case TelemetryType.Event:
+                            this.eventFilterGroups.Add(new FilterConjunctionGroup<EventTelemetry>(documentFilterConjunctionGroupInfo.Filters, out groupErrors));
+                            break;
+                        case TelemetryType.Trace:
+                            this.traceFilterGroups.Add(new FilterConjunctionGroup<TraceTelemetry>(documentFilterConjunctionGroupInfo.Filters, out groupErrors));
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException(string.Format(CultureInfo.InvariantCulture, "Unsupported TelemetryType: '{0}'", documentFilterConjunctionGroupInfo.TelemetryType));
+                    }
+
+                    errorList.AddRange(groupErrors);
+                }
+                catch (Exception e)
```

Comment:
Either something broke completely and we get an exception from the switch's default clause, or there's something wrong with configuration and we get an exception from FilterConjunctionGroup's constructor which parses user-supplied filter configuration. I don't care which exception it is, my reaction is to report it up the call chain, hence catch(Exception).